### ISSUE TITLE
Sparsemax

### ISSUE THREAD

**Issue by i55code** [awslabs/sockeye#865]
Hi Sockeye team, if I want to change the softmax layer to sparsemax or other customized -maxes, may you point me which part of the code to start with? Thanks!

**Comment by fhieber** [comment]
Hi @i55code,
you would probably want to take a look at https://github.com/awslabs/sockeye/blob/master/sockeye/loss.py which defines the Cross-Entropy/Softmax losses during training. At inference, this is the part where we apply `log_softmax` to the logits from the decoder: https://github.com/awslabs/sockeye/blob/master/sockeye/beam_search.py#L77

**Comment by i55code** [comment]
Thank you so much. This is very helpful!
